enumerables.rb

```diff
@@ -0,0 +1,108 @@
+module Enumerable
+  def my_each
+    length.times { |i| yield(self[i]) }
+  end
+
+  def my_each_with_index
+    (0..length - 1).each do |index|
+      yield(self[index], index)
+    end
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| break result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| break result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| break result = false if yield(i) }
+    result
+  end
+
+  def my_count(argument)
+    counter = 0
+    my_each { |i| counter += 1 if argument == i }
+    counter
+  end
+
+  def my_map(proc = nil)
+    result = []
+    my_each do |item|
+      result << if block_given?
+                  yield(item)
+                else
+                  proc.call(item)
+                end
+    end
+    result
+  end
+
+  def my_inject(initial = nil)
+    result = initial.nil? ? self[0] : initial
+    index = initial.nil? ? 1 : 0
+    self[index...length].my_each do |item|
+      result = yield(result, item)
+    end
+    result
+  end
+end
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
(5..50).my_inject(4) { |prod, n| prod * n } != (5..50).inject(4) { |prod, n| prod * n }

[1,2,3,4,5].my_inject(:+) != [1,2,3,4,5].inject(:+)
```